Added support for scrolller.com

PaaaulZ opened the pull request:
# Category

This change is exactly one of the following (please change `[ ]` to `[x]`) to indicate which:
* [ ] a bug fix (Fix #...)
* [x] a new Ripper
* [ ] a refactoring
* [ ] a style change/fix
* [ ] a new feature


# Description

Adds support for scrolller.com (requested in issue #1814 ).  Just some notes:

1) I could not find a way with jsoup and the functions provided by ripme to make a post request with a json body so I used another method with HTTPUrlConnection. Seems like jsoup supports only key/value pair and does not support changing the entire request body (or I'm unable to), I even try empty value and body in the key.
2) I could not find a way to make a websocket connection without adding dependencies. Unfortunately scrolller.com uses POST requests for normal API requests but uses WebSockets if you choose the sort parameter, this means that the ripper won't support sorting. Sad news is that the site's default sorting is Random, so we can only rip random posts...

Let me know if you have suggestions (especially for point 2), I would like to improve this.

# Testing

Required verification:
* [x] I've verified that there are no regressions in `mvn test` (there are no new failures or errors).
* [x] I've verified that this change works as intended.
  * [x] Downloads all relevant content.
  * [x] Downloads content from multiple pages (as necessary or appropriate).
  * [x] Saves content at reasonable file names (e.g. page titles or content IDs) to help easily browse downloaded content.
* [x] I've verified that this change did not break existing functionality (especially in the Ripper I modified).

Optional but recommended:
* [x] I've added a unit test to cover my change.

soloturn commented:
if it is easier why not adding dependencies? currently the unit test fails:
```
Error:    ScrolllerRipperTest.testScrolllerFilterRegex:48 expected: <> but was: <NOFILTER>
```

PaaaulZ commented:
I thought it was best making a pr for a 90% completed ripper and ask for approval, it didn't feel "right" to add an entire dependency just for one ripper but if it's ok for you I'll find one and work on it.
Also i fixed the test. I'm pretty sure it worked when I opened the issue but maybe I forgot to push when I changed "" to "NOFILTER", my bad.

I'll find a dependency for WebSocket, add support for the "sort" parameter and push as soon as I can so you can review it.
Thanks for now

PaaaulZ commented:
Added dependency and support for the "sort" parameter, now the ripper is complete.
Code can be better, there is some duplicated code in "getURLsFromJSON" that I left because it looked cleaner than a chain of if/else statements and other small stuff.

Let me know if you have any suggestions on how to improve it or if something does not add up

rautamiekka commented:
> Added dependency and support for the "sort" parameter, now the ripper is complete.
> Code can be better, there is some duplicated code in "getURLsFromJSON" that I left because it looked cleaner than a chain of if/else statements and other small stuff.
> 
> Let me know if you have any suggestions on how to improve it or if something does not add up

DRY is far more important than "cleaner" when you'll have to update 2 sets of 98-99% identical code. This is a great way to get nasty bugs.

PaaaulZ commented:
Removed duplicates around code

soloturn commented:
included in
https://github.com/ripmeapp2/ripme/actions

mukulj77 commented:
@PaaaulZ Mate, thanks a lot for the feature. Scrolller has some great stuff and it's great that you added support for it. 
I'll go through your implementation after couple of weeks. I look forward to learn from your work :) 

Cheers.
